google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplWithTransactionOptionsTest.java

```
@@ -0,0 +1,497 @@
+    client
+        .readWriteTransaction(Options.serializableIsolationLevel())
+        .run(
+            transaction -> {
+              try (ResultSet rs = transaction.executeQuery(INVALID_SELECT_STATEMENT)) {
+                SpannerException e = assertThrows(SpannerException.class, () -> rs.next());
+                assertEquals(ErrorCode.INVALID_ARGUMENT, e.getErrorCode());
+              }
+              return transaction.executeUpdate(UPDATE_STATEMENT);
+            });
+    validateIsolationLevel(IsolationLevel.SERIALIZABLE);
+  }
+
+  private void validateIsolationLevel(IsolationLevel isolationLevel) {
+    boolean foundMatchingRequest = false;
+    for (AbstractMessage request : mockSpanner.getRequests()) {
+      if (request instanceof ExecuteSqlRequest) {
+        foundMatchingRequest = true;
+        assertEquals(
+            ((ExecuteSqlRequest) request).getTransaction().getBegin().getIsolationLevel(),
+            isolationLevel);
+      } else if (request instanceof BeginTransactionRequest) {
+        foundMatchingRequest = true;
+        assertEquals(
+            ((BeginTransactionRequest) request).getOptions().getIsolationLevel(), isolationLevel);
+      } else if (request instanceof ReadRequest) {
+        foundMatchingRequest = true;
+        assertEquals(
+            ((ReadRequest) request).getTransaction().getBegin().getIsolationLevel(),
+            isolationLevel);
+      } else if (request instanceof CommitRequest) {
+        foundMatchingRequest = true;
+        assertEquals(
+            ((CommitRequest) request).getSingleUseTransaction().getIsolationLevel(),
+            isolationLevel);
+      } else if (request instanceof ExecuteBatchDmlRequest) {
+        foundMatchingRequest = true;
+        assertEquals(
+            ((ExecuteBatchDmlRequest) request).getTransaction().getBegin().getIsolationLevel(),
+            isolationLevel);
+      }
+      if (foundMatchingRequest) {
+        break;
+      }
+    }
```

Can we throw an error if no matching request found? Ideally there should be atleast. For some reason if gRPC call itself is not made, then it should throw an error

---

Added